packages/mui-material-next/src/InputLabel/InputLabel.tsx

```diff
@@ -88,17 +88,61 @@ const InputLabelRoot = styled(FormLabel, {
     pointerEvents: 'none',
     transform: 'translate(16px, 16px) scale(1)',
     maxWidth: 'calc(100% - 24px)',
+    // adjust maxWidth by 40px per adornment: adornment width 24px + inner padding 16px = 40px
+    // adornedStart only
+    ...(ownerState.formControl?.adornedStart &&
+      !ownerState.formControl?.adornedEnd && {
+        transform: 'translate(52px, 16px) scale(1)',
+        maxWidth: 'calc(100% - 24px - 40px)',
+      }),
+    // adornedEnd only
+    ...(ownerState.formControl?.adornedEnd &&
+      !ownerState.formControl?.adornedStart && {
+        maxWidth: 'calc(100% - 24px - 40px)',
+      }),
+    // adornedStart and adornedEnd
+    ...(ownerState.formControl?.adornedStart &&
+      ownerState.formControl?.adornedEnd && {
+        transform: 'translate(52px, 16px) scale(1)',
+        maxWidth: 'calc(100% - 24px - 40px - 40px)',
+      }),
     ...(ownerState.size === 'small' && {
       transform: 'translate(12px, 13px) scale(1)',
+      ...(ownerState.formControl?.adornedStart && {
+        transform: 'translate(52px, 13px) scale(1)',
+      }),
     }),
     ...(ownerState.shrink && {
       userSelect: 'none',
       pointerEvents: 'auto',
       transform: 'translate(16px, 7px) scale(0.75)',
-      maxWidth: 'calc(133% - 24px)',
+      maxWidth: 'calc(133% - 32px)',
       ...(ownerState.size === 'small' && {
-        transform: 'translate(12px, 4px) scale(0.75)',
+        transform: 'translate(16px, 4px) scale(0.75)',
       }),
+      // adornedStart only
+      ...(ownerState.formControl?.adornedStart &&
+        !ownerState.formControl?.adornedEnd && {
+          transform: 'translate(52px, 7px) scale(0.75)',
+          maxWidth: 'calc(133% - 32px - 48px - 8px)',
```

Comment:
When it's in the `shrink` state, the correct maxWidth adjustment seems to be "adjust by 48px per adornment, then adjust by another 8px", though I'm not 100% sure if this is a real "pattern" or if my mind is just trying to fill in the gaps 😅